File: cems-SPA/src/app/_models/CemsLog.model.ts
import { ExceptionDetails } from './ExceptionDetails.model';
import { TrackedState } from './TrackedState.enum';

export class CemsLog {
    public id: number;
    public platform: number;
    public timestamp: Date;
    public exceptionDetails: ExceptionDetails;

    public createdTime: Date;
    public stateChangedTime: Date;
    public currentState: TrackedState;

    constructor(logObj) {
        this.id = logObj.id;
        this.platform = logObj.platform;
        this.timestamp = new Date(logObj.timestamp);
        this.createdTime = new Date(logObj.createdTime);
        this.stateChangedTime = new Date(logObj.stateChangedTime);
        this.currentState = logObj.currentState;
        this.exceptionDetails = new ExceptionDetails(logObj.exceptionDetails);
    }
}

File: cems.API/Models/CemsLogModel.cs
using System;
using cems.API.Models.user;
using Newtonsoft.Json;

namespace cems.API.Models
{
    public class CemsLogModel : ITrackable
    {
        public int Id { get; set; }
        public Platforms Platform { get; set; }
        public DateTime Timestamp { get; set; }
        public CemsExceptionDetailsModel ExceptionDetails { get; set; }
        public CemsLogModel() { }
        public CemsLogModel(Platforms platform)
        {
            Platform = platform;
            ExceptionDetails = new CemsExceptionDetailsModel();
            Timestamp = DateTime.Now;
        }

        [JsonIgnore] public int WebApiKeyId { get; set; }
        [JsonIgnore] public WebApiKey WebApiKey { get; set; }

        public DateTime CreatedTime { get; set; }
        public DateTime StateChangedTime { get; set; }
        public TrackedState CurrentState { get; set; }
    }
}
